app/controllers/fhir/r4/api_controller.rb

```diff
@@ -424,6 +424,120 @@ def all
+
+    # Create the download to uniquely id this request, and queue the ExportFhirJob
+    download = ApiDownload.create(application_id: client_app.id, url: request.url)
+    export_job = ExportFhirJob.perform_later(client_app, download, { since: since })
+    # Add the job_id of the job to the download for reference in status requests
+    download.update(job_id: export_job.provider_job_id)
+
+    respond_to do |format|
+      format.any { head(:accepted, content_location: "#{root_url}fhir/r4/ExportStatus/#{download.id}") }
+    end
+  end
+
+  # Give an update on the status of a bulk FHIR data request
+  #
+  # GET /fhir/r4/ExportStatus/[:id]
+  def export_status
+    id = params.permit(:id)[:id]
+    download = ApiDownload.find_by_id(id)
+
+    case Sidekiq::Status.status(download&.job_id)
+    when :complete
+      response_json = {
+        transactionTime: Sidekiq::Status.get(download.job_id, :transaction_time),
+        request: download.url,
+        requiresAccessToken: true,
+        output: ActiveStorage::Blob.where(id: download.files.pluck(:blob_id)).pluck(:filename).map do |file|
+          {
+            type: file.split('.').first,
+            url: "#{root_url}fhir/r4/ExportFiles/#{id}/#{file}"
+          }
+        end
+      }
+      render json: response_json.to_json, status: :ok
+    when :failed
+      status_server_error(['Export failed'])
+    when :working, :queued, :retrying
+      head :accepted
+    else
+      status_not_found_with_custom_errors(['No export found for this ID'])
+    end
+  end
+
+  # Return an bulk exported ndjson file
+  #
+  # GET /ExportFiles/[:id]/[:filename]
+  def export_files
+    id = params.permit(:id)[:id]
+    filename = params.permit(:filename)[:filename]
+    resource_type = filename&.split('.')&.first&.downcase
+
+    case resource_type
+    when 'patient'
+      return if doorkeeper_authorize!(*PATIENT_READ_SCOPES)
+    when 'observation'
+      return if doorkeeper_authorize!(*OBSERVATION_READ_SCOPES)
+    when 'questionnaireresponse'
+      return if doorkeeper_authorize!(*QUESTIONNAIRE_RESPONSE_READ_SCOPES)
+    when 'relatedperson'
+      return if doorkeeper_authorize!(*RELATED_PERSON_READ_SCOPES)
+    when 'immunization'
+      return if doorkeeper_authorize!(*IMMUNIZATION_READ_SCOPES)
+    when 'provenance'
+      return if doorkeeper_authorize!(*PROVENANCE_READ_SCOPES)
+    else
+      status_not_found_with_custom_errors(["Invalid ResourceType '#{resource_type}'"]) && return
+    end
+
+    download = current_client_application&.api_downloads&.find_by_id(id)
+    status_not_found_with_custom_errors(['No file found at this URL']) && return if download.nil?
+
+    # Set the headers before streaming the returned content
+    response.headers['Content-Type'] = 'application/fhir+ndjson'
+    # Find and return the specific attachment associated with the given filename
+    ActiveStorage::Blob.where(id: download.files.pluck(:blob_id)).find_by(filename: filename)&.download do |chunk|
```

Comment:
After looking at this a while I think a better way to access the `ActiveStorage::Blob` would be through the public ActiveStorage API and Attached Many proxy instead of going right to the Blob.

Something like `ApiDownload.find(id).files.blobs.where(filename: filename).first.download`

This can also replace the usage on around [L486](https://github.com/SaraAlert/SaraAlert/pull/987/files#diff-e79e38176ee697a8541cffba23c60d9decaa69382ca5c6e875da033508e5b428R486)

Reply:
Fixed in https://github.com/SaraAlert/SaraAlert/pull/987/commits/6977cd314b84a7c5ecfd8d2d2ac200bf5d218e7b.